### DISCUSSION TITLE
Nonsensical ordering of suggestion tables

### DISCUSSION THREAD

**Discussion by givihuda** [dbeaver/dbeaver#17988] [General]
This kind of thing has thrown me off when typing, causing the wrong table to be placed by the time I hit Enter when I don't realize the wrong one ended up at the top:

![image](https://user-images.githubusercontent.com/74785256/197047727-c32d2aff-99ab-4859-b959-4f5d4af759b5.png)

So I'm typing `JOIN ord`, the list pops up with `order` suggestion at first, but ultimately `order_type` ends up at the top...

Here's the thing... even if they're switching places for some reason, the suggestions are just ordered nonsensibly when it comes to a word without the underscore and another word with a following underscore:

`order`
`order_type`

That makes sense; `order` has 5 characters and no subsequent underscore. `order_type` has more than 5 characters, and an underscore following the first 5. Why would `order_type` be ordered above `order`?! That makes no sense. The 2nd name includes the whole of the first and then adds characters. 

How is _nothing_ (not even a space character) considered to be ordered _after_ **_something_** (the underscore)? No matter where underscore is on the ASCII list, it's still _an additional character_ that the `order` doesn't have... the 6th character of `order` has no ASCII value (essentially null) because there _is no 6th character_. I'd think _no ASCII value present for this slot_ would be ordered before _an ASCII value present for this slot_.

Can someone explain why this is happening? Or how it gets fixed? Even if this is just "Well it's defaulting to whatever underneath like a 3rd-party package or the OS", it's still _nonsensical_ behavior, and shouldn't be in DBeaver even if it's in some underlying or side-loaded software. DBeaver should be better, it should behave sensibly for users trying to utilize its features, like suggestions.

**Comment by LonwoLonwo** [comment]
Hello @givihuda 

What about this setting? Is it enabled?

![2022-10-27 23_53_17-AnyDesk](https://user-images.githubusercontent.com/45152336/198396237-4eb57c3e-9e3a-4580-bf98-71cde90ef478.png)

**Comment by givihuda** [comment]
That setting was already enabled...

![image](https://user-images.githubusercontent.com/74785256/199058645-df8a2ba6-cef3-4b80-acbf-3e825222ffff.png)

So... the issue is that DBeaver is sorting alphabetically in its own nonsensical way.

Here's how Mac Finder sorts columns Name and Kind for me:

![image](https://user-images.githubusercontent.com/74785256/199058817-42a7cd1a-0ce3-4eae-8355-bfe4b5f76075.png)

Here's how ModernCSV automatically sorts the file names:

![image](https://user-images.githubusercontent.com/74785256/199058973-5500a898-a088-42be-9bb5-60e55151152e.png)

Yes I know I'm comparing an apple to an orange, but we're talking about text sorting (like fruit sorting)... other apps that I check sort text alphabetically sensibly; DBeaver does not. 

If it's inheriting text sorting from Mac logic... well, it's the only app I'm seeing that's doing it differently than even the Mac does with Finder.

So what is DBeaver basing its alphabetical sorting logic on? I'm not finding a lexical standard for this yet, nor why it's implemented, if it's not a bug...
